modules/runner/src/main/java/org/apache/ignite/runner/internal/app/IgnitionImpl.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.ignite.internal.affinity.AffinityManager;
+import org.apache.ignite.configuration.storage.ConfigurationStorage;
+import org.apache.ignite.internal.table.distributed.TableManagerImpl;
+import org.apache.ignite.internal.vault.VaultManager;
+import org.apache.ignite.internal.metastorage.MetaStorageManager;
+import org.apache.ignite.network.ClusterLocalConfiguration;
+import org.apache.ignite.network.ClusterService;
+import org.apache.ignite.network.message.MessageSerializationRegistry;
+import org.apache.ignite.network.scalecube.ScaleCubeClusterServiceFactory;
+import org.apache.ignite.raft.internal.Loza;
+import org.apache.ignite.runner.internal.storage.DistributedConfigurationStorage;
+import org.apache.ignite.runner.internal.storage.LocalConfigurationStorage;
+import org.apache.ignite.internal.schema.SchemaManager;
+import org.apache.ignite.table.manager.TableManager;
+import org.apache.ignite.utils.IgniteProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of an entry point for handling grid lifecycle.
+ */
+public class IgnitionImpl implements Ignition {
+    /** */
+    private static final String[] BANNER = new String[] {
+        "",
+        "           #              ___                         __",
+        "         ###             /   |   ____   ____ _ _____ / /_   ___",
+        "     #  #####           / /| |  / __ \\ / __ `// ___// __ \\ / _ \\",
+        "   ###  ######         / ___ | / /_/ // /_/ // /__ / / / // ___/",
+        "  #####  #######      /_/  |_|/ .___/ \\__,_/ \\___//_/ /_/ \\___/",
+        "  #######  ######            /_/",
+        "    ########  ####        ____               _  __           _____",
+        "   #  ########  ##       /  _/____ _ ____   (_)/ /_ ___     |__  /",
+        "  ####  #######  #       / / / __ `// __ \\ / // __// _ \\     /_ <",
+        "   #####  #####        _/ / / /_/ // / / // // /_ / ___/   ___/ /",
+        "     ####  ##         /___/ \\__, //_/ /_//_/ \\__/ \\___/   /____/",
+        "       ##                  /____/\n"
+    };
+
+    /** */
+    private static final String VER_KEY = "version";
+
+    /** */
+    private static final Logger log = LoggerFactory.getLogger(IgnitionImpl.class);
+
+    /** {@inheritDoc} */
+    @Override public synchronized Ignite start(String jsonStrBootstrapCfg) {
+        ackBanner();
+
+        // Vault Component startup.
+        VaultManager vaultMgr = new VaultManager();
+
+        boolean cfgBootstrappedFromPds = vaultMgr.bootstrapped();
+
+        List<RootKey<?, ?>> rootKeys = new ArrayList<>(Collections.singletonList(NetworkConfiguration.KEY));
+
+        List<ConfigurationStorage> configurationStorages =
+            new ArrayList<>(Collections.singletonList(new LocalConfigurationStorage(vaultMgr)));
+
+        // Bootstrap local configuration manager.
+        ConfigurationManager locConfigurationMgr = new ConfigurationManager(rootKeys, configurationStorages);
```

Comment:
locConfigurationMgr is a subset of configurationMgr. Can we somehow have only 1 ConfigurationManager ? Maybe distributed store can be registered later as soon as it's will be ready ?

Reply:
> Maybe distributed store can be registered later as soon as it's will be ready
Initially It was implemented that way, however it means that we have circular reference. In order to eliminate such reference locConfigurationManager was introduced.
So, as a resutl we have following dependencies:

- ConfigurationManager depends on localConfigurationStorage and distributedConfigurationStorage.
- MetaStorageManager is a provider for distirbutedConfigurationStorage and depends on Network, Volt and RaftManager. It could also depend on locConfigurationManger.
- RaftManager depends on Network and possible locConfigurationManager.
- Network depends on locConfigurationManager.
- locConfigurationManager depends on locConfigurationStorage that is provided by Vault.

For full dependency graph and more details please see https://cwiki.apache.org/confluence/display/IGNITE/IEP-73%3A+Node+startup